lib/sles4sap_publiccloud.pm

```diff
@@ -831,4 +837,105 @@ sub list_cluster_nodes {
     die 'None of the hosts are currently part of existing cluster' unless @cluster_nodes;
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online, 0 otherwise.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self, $password_db, $instance_id) = @_;
+    my $hdb_cmd = "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'";
+
+    sleep 30;
+    eval {
+        my $output_cmd = $self->run_cmd(cmd => $hdb_cmd, runas => get_required_var("SAP_SIDADM"));
+    };
+    if ($@) {
+        record_info('HANA DB OFFLINE', "Hana database in primary node is offline. Here the error \n$@");
+        return 0;
+    }
+    else {
+        return 1;
+    }
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns 1 if hana database is online, 0 if offline
+
+=over 1
+
+=item B<TIMEOUT> - default 900
+=item B<TOTAL_CONSECUTIVE_PASSES> - default 5
+
+=back
+=cut
+
+sub is_hana_database_offline {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 900);
+    my $total_consecutive_passes = ($args{total_consecutive_passes} // 5);
+    my $instance_id = get_required_var('INSTANCE_ID');
+
+    my $db_status = -1;
+    my $consecutive_passes = 0;
+    my $password_db = "Linux1234";
```

Comment:
DB password should not be hard coded. What happens if DB is installed with a different password? Then this test will fail. Please check where the installer is getting the DB password from, and do the same here.